pkg/controller/download_request_controller_test.go

```diff
@@ -18,310 +18,256 @@ package controller
 	"github.com/vmware-tanzu/velero/pkg/builder"
-	"github.com/vmware-tanzu/velero/pkg/generated/clientset/versioned/fake"
-	informers "github.com/vmware-tanzu/velero/pkg/generated/informers/externalversions"
 	"github.com/vmware-tanzu/velero/pkg/persistence"
 	persistencemocks "github.com/vmware-tanzu/velero/pkg/persistence/mocks"
 	"github.com/vmware-tanzu/velero/pkg/plugin/clientmgmt"
 	pluginmocks "github.com/vmware-tanzu/velero/pkg/plugin/mocks"
 	velerotest "github.com/vmware-tanzu/velero/pkg/test"
-	kubeutil "github.com/vmware-tanzu/velero/pkg/util/kube"
 )
 
-type downloadRequestTestHarness struct {
-	client          *fake.Clientset
-	informerFactory informers.SharedInformerFactory
-	pluginManager   *pluginmocks.Manager
-	backupStore     *persistencemocks.BackupStore
-
-	controller *downloadRequestController
-}
-
-func newDownloadRequestTestHarness(t *testing.T, fakeClient client.Client) *downloadRequestTestHarness {
-	var (
-		client          = fake.NewSimpleClientset()
-		informerFactory = informers.NewSharedInformerFactory(client, 0)
-		pluginManager   = new(pluginmocks.Manager)
-		backupStore     = new(persistencemocks.BackupStore)
-		controller      = NewDownloadRequestController(
-			client.VeleroV1(),
-			informerFactory.Velero().V1().DownloadRequests(),
-			informerFactory.Velero().V1().Restores().Lister(),
-			fakeClient,
-			informerFactory.Velero().V1().Backups().Lister(),
-			func(logrus.FieldLogger) clientmgmt.Manager { return pluginManager },
-			velerotest.NewLogger(),
-		).(*downloadRequestController)
-	)
-
-	clockTime, err := time.Parse(time.RFC1123, time.RFC1123)
-	require.NoError(t, err)
-	controller.clock = clock.NewFakeClock(clockTime)
-
-	controller.newBackupStore = func(*velerov1api.BackupStorageLocation, persistence.ObjectStoreGetter, logrus.FieldLogger) (persistence.BackupStore, error) {
-		return backupStore, nil
-	}
-
-	pluginManager.On("CleanupClients").Return()
-
-	return &downloadRequestTestHarness{
-		client:          client,
-		informerFactory: informerFactory,
-		pluginManager:   pluginManager,
-		backupStore:     backupStore,
-		controller:      controller,
-	}
-}
-
-func newDownloadRequest(phase velerov1api.DownloadRequestPhase, targetKind velerov1api.DownloadTargetKind, targetName string) *velerov1api.DownloadRequest {
-	return &velerov1api.DownloadRequest{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "a-download-request",
-			Namespace: velerov1api.DefaultNamespace,
-		},
-		Spec: velerov1api.DownloadRequestSpec{
-			Target: velerov1api.DownloadTarget{
-				Kind: targetKind,
-				Name: targetName,
+var _ = Describe("Download Request Reconciler", func() {
+	BeforeEach(func() {})
+	AfterEach(func() {})
+
+	It("Should successfully patch a download request object status phase", func() {
+		defaultBackup := func() *velerov1api.Backup {
+			return builder.ForBackup(velerov1api.DefaultNamespace, "a-backup").StorageLocation("a-location").Result()
+		}
+
+		tests := []struct {
```

Comment:
I think if we're going to use Ginkgo here, we should be using the `DescribeTable` function for these tests as you did in #3180. I think without that, all these tests are treated as one spec and you don't get to see the name of a test if it fails.

Reply:
Done.